src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -66,43 +66,69 @@
   public static final String DTD_FILE_NAME = "game.dtd";
   private final String mapName;
 
-  public GameParser(final String mapName) {
-    this.mapName = Preconditions.checkNotNull(mapName);
+  private GameParser(final String mapName) {
+    this.mapName = mapName;
   }
 
   /**
-   * Parses the XML Document provided by the given InputStream into a DOM.
-   * 
-   * @return The root Element of the DOM
+   * Performs a deep parse of the game definition contained in the specified stream.
+   *
+   * @return A complete {@link GameData} instance that can be used to play the game.
    */
-  private Element parseDom(final InputStream stream) throws SAXException {
-    Preconditions.checkNotNull(stream, "InputStream must be non-null!");
-    return getDocument(stream).getDocumentElement();
+  public static GameData parse(final String mapName, final InputStream stream)
+      throws GameParseException, SAXException, EngineVersionException {
+    return parse(mapName, stream, new AtomicReference<>());
   }
 
   /**
-   * Parses a file into a GameData object.
+   * Performs a deep parse of the game definition contained in the specified stream.
+   *
+   * @param gameName Receives the game name parsed from the game definition.
+   *
+   * @return A complete {@link GameData} instance that can be used to play the game.
    */
-  public GameData parse(final InputStream stream, final AtomicReference<String> gameName)
+  public static GameData parse(final String mapName, final InputStream stream, final AtomicReference<String> gameName)
```

Comment:
Unrelated to this PR, but passing an empty reference looks like a code smell to me

Reply:
Agreed.  Actually, if you look deeper down the call stack, the code that ultimately receives this reference assumes the caller will specify `null` if they aren't interested in receiving the value.  But passing `null` is just as bad of a smell, IMO. :smile: